app/controllers/heritages_controller.rb

```diff
@@ -151,6 +151,20 @@ def load_district
     @district = District.find_by!(name: params[:district_id])
   end
 
+  def retrieve_manifest
+    heritage = Heritage.find_by!(name: params[:heritage_id])
+    user = params[:user]
+    password = params[:password]
+
+    if heritage.image_name.start_with?("quay.io/")
+      response = RetrieveQuayManifest.new().process(user, password, heritage)
+    else
+      raise ExceptionHandler::InternalServerError.new("Unsupported Registry")
+    end
```

Comment:
This is strange because we should be able to login to quay using the docker API. Please see the docker CLI implementation https://github.com/docker/cli/blob/a32cd16160f1b41c1c4ae7bee4dac929d1484e59/cli/command/registry/login.go

It is possible for the docker CLI to login to quay without having to use the quay-specific CLI, which I think should be the case with barcelona as well.

Reply:
@davidsiaw  
Sure, we can use the docker api to `log in`.
but  to use `manifest` endpint `https://docs.docker.com/registry/spec/api/#manifest`
we need bare token  and it requires a `basic auth`
I do not think docker CLI support `manifest` endpoint yet.

```
curl -X GET --user username:password "https://quay.io/v2/auth?service=quay.io&scope=repository:NAMESPACE/REPOSITORY:pull"
```
I think once https://docs.docker.com/engine/reference/commandline/manifest_inspect/ is not experimental,
we can start to look how they are doing it.